### PR TITLE
Remove usage of initializers and adapt PV-labelling logic to prepare for K8s 1.14

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
Due to kubernetes/kubernetes#72972 the alpha `Initializers` feature will be entirely removed with Kubernetes 1.14. We are currently relying on it as the cloud-controller-manager uses it to label persistent volumes before they become visible in the API.
This functionality is also removed (see kubernetes/kubernetes#73102) with 1.14.
As a consequence, in order to use PV labelling until the community has migrated all CCMs to use `MutatingWebhookConfiguration`s for that, the only way is to enable the (already deprecated) `PersistentVolumeLabel` admission controller in the kube-apiserver.
That means that the kube-apiserver needs to talk to the cloud provider to figure out the region/zone of the PV (and later add them as label).
--> We have to provide the cloud credentials + config to the API server in this case.

Shoot clusters with Kubernetes version < 1.14 still have an enabled initializer feature.

**Special notes for your reviewer**:
The kube-apiserver does now talk to the cloud provider to figure out information about created PVs. However, it **does not** use the cloud's metadata service but its public API. We don't have to change our network policy preventing the kube-apiserver's from accessing the metadata service.

/cc @marwinski @ThormaehlenFred 

**Release note**:

```noteworthy operator
Due to the fact that the `Initializer` feature is entirely dropped with Kubernetes 1.14 Gardener does now no longer rely on it. It was used to enable the cloud-controller-manager for labelling created PVs. This is now done by the `PersistentVolumeLabel` admission controller inside the kube-apiserver. Consequently, the kube-apiserver does now have information about the cloud (credentials/config).
```

### CODE REVIEW

**Comment by mvladev** [comment]
What removes existing deployed resources - controller, mutatingwebhooks and so?

I would suggest to keep the templates and use them to do clean up. After 1 or more release we can remove them as well.

**Comment by rfranzke** [comment]
Which concrete existing deployed resources are you referring to?

**Comment by mvladev** [comment]
Those 2:
https://github.com/gardener/gardener/blob/0949217791253338d1a908840b85c3656c6135e9/charts/shoot-core/charts/cloud-controller-manager/templates/rbac-pvl-controller.yaml
https://github.com/gardener/gardener/blob/0949217791253338d1a908840b85c3656c6135e9/charts/shoot-core/charts/cloud-controller-manager/templates/webhook-gardener-add-initializer.yaml

**Comment by rfranzke** [comment]
Not needed. kube-addon-manager will take care removing them.

**Comment by mvladev** [comment]
Oh yes - they are deployed in the Shoot. What about:
https://github.com/gardener/gardener/blob/0949217791253338d1a908840b85c3656c6135e9/charts/seed-bootstrap/templates/gardener-external-admission-controller.yaml

**Comment by rfranzke** [comment]
https://github.com/gardener/gardener/pull/838/files#diff-5ca37df1e48dc263572bf7822db58a8aR339

File: pkg/client/kubernetes/admissionplugins_test.go [hunk 1]
```diff
@@ -63,5 +63,15 @@ var _ = Describe("kubernetes", func() {
 				Expect(plugins).To(ContainElement(gardenv1beta1.AdmissionPlugin{Name: plugin}))
 			}
 		})
+
+		It("should return the list for 1.14 or higher", func() {
+			expected := []string{"Priority", "NamespaceLifecycle", "LimitRanger", "ServiceAccount", "NodeRestriction", "DefaultStorageClass", "DefaultTolerationSeconds", "ResourceQuota", "StorageObjectInUseProtection", "MutatingAdmissionWebhook", "ValidatingAdmissionWebhook"}
+
+			plugins := GetAdmissionPluginsForVersion("1.11.23")
```
Comment (timuthy) [inline pkg/client/kubernetes/admissionplugins_test.go]:
```suggestion
			plugins := GetAdmissionPluginsForVersion("1.14.0")
```

**Review by timuthy — CHANGES_REQUESTED** [review]

**Review by timuthy — DISMISSED** [review]
/lgtm

**Review by adracus — APPROVED** [review]

**Pull request merged**